### PR TITLE
Parent child layout refactor

### LINKED ISSUE #163: avoid motion if reduced motion is on in parent-child layout
motion styles should be wrapped in `@media (prefers-reduced-motion: no-preference)` in `ParentChildLayout`

### PR DESCRIPTION
**Pull request by lukeac123**
Closes #163

- removes parentPosition and disableAnimation props
- collapsedViewElement renamed to collapsedView
- reduced motion changes animation to fade in, instead of slide in
- refactored useIsViewportLargerThanBreakpoint hook
- reworked cy tests
- Site docs probably need a bit of an update, will do in another pr

### CODE REVIEW

File: packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx
Comment (lukeac123) [inline packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx:1]: test for onCollapseChange

File: packages/lab/src/parent-child-layout/ParentChildLayout.css [hunk 1]
```diff
@@ -1,30 +1,30 @@
 .saltParentChildLayout {
-  overflow: hidden;
+  display: flex;
+  gap: calc(var(--parentChildLayout-gap) * 3);
 }
 
-.saltParentChildLayout-parent {
-  animation: var(--salt-animation-slide-in-left);
+.saltParentChildLayout-child {
+  flex-grow: 1;
```
Comment (lukeac123) [inline packages/lab/src/parent-child-layout/ParentChildLayout.css:7]: Child will expand to fill the rest of the container - can be seen in the preferences dialog example.
Think this makes the most sense if the parent is a fixed width or leaving empty space. Use can add styling to align centre or fix the total width of the ParentChildLayout if preferred

File: packages/lab/src/parent-child-layout/ParentChildLayout.css [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
 }
 
-.saltParentChildLayout-child {
-  animation: var(--salt-animation-slide-in-right);
+.saltParentChildLayout-collapsed {
+  width: 100%;
```
Comment (lukeac123) [inline packages/lab/src/parent-child-layout/ParentChildLayout.css:11]: when collapsed - parent / child should fill the container

File: packages/lab/src/parent-child-layout/ParentChildLayout.css [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
 }
 
-.saltParentChildLayout-reversed.saltFlexLayout {
-  --flexLayout-direction: row-reverse;
```
Comment (lukeac123) [inline packages/lab/src/parent-child-layout/ParentChildLayout.css:14]: feature not in scope for v1

File: packages/lab/src/parent-child-layout/ParentChildLayout.tsx [hunk 4]
```diff
@@ -1,39 +1,33 @@
...
   /**
    * Change element that is displayed when in staked view.
    */
-  collapsedViewElement?: StackedViewElement;
+  collapsedView?: "child" | "parent";
   /**
-   * Disable all animations.
+   * Controls the space between parent and child components, default is 0.
    */
-  disableAnimations?: boolean;
```
Comment (lukeac123) [inline packages/lab/src/parent-child-layout/ParentChildLayout.tsx:32]: reduced motion can be used - removed as not needed

File: packages/lab/src/parent-child-layout/ParentChildLayout.tsx [hunk 5]
```diff
@@ -1,39 +1,33 @@
...
-  /**
-   * Position of the parent component within the layout.
-   */
-  parentPosition?: "left" | "right";
```
Comment (lukeac123) [inline packages/lab/src/parent-child-layout/ParentChildLayout.tsx:36]: not in scope for v1

File: packages/lab/src/utils/useIsViewportLargerThanBreakpoint.ts [hunk 6]
```diff
@@ -9,8 +9,7 @@ export const useIsViewportLargerThanBreakpoint = (
 ): boolean => {
   const orderedBreakpoints = useOrderedBreakpoints();
   const index = orderedBreakpoints.indexOf(targetedBreakpoint);
-  const allPreviousBreakpoints = orderedBreakpoints.slice(0, index + 1);
-
   const currentBreakpoint = useCurrentBreakpoint();
-  return allPreviousBreakpoints.includes(currentBreakpoint);
```
Comment (lukeac123) [inline packages/lab/src/utils/useIsViewportLargerThanBreakpoint.ts:15]: Not sure why checking to see if the current breakpoint is in the allPreviousBreakpoints is needed, so changed to a direct comparison of the index >= the current breakpoint

File: packages/lab/src/parent-child-layout/ParentChildLayout.tsx [hunk 7]
```diff
@@ -1,39 +1,33 @@
...
+  useEffect,
+} from "react";
 import { useIsViewportLargerThanBreakpoint } from "../utils";
-
 import { useWindow } from "@salt-ds/window";
 import { useComponentCssInjection } from "@salt-ds/styles";
 
 import parentChildLayoutCss from "./ParentChildLayout.css";
 
 export type StackedViewElement = "parent" | "child";
```
Comment (Fercas123) [inline packages/lab/src/parent-child-layout/ParentChildLayout.tsx:15]: you can probably remove the type if it is not going to be used

File: packages/lab/stories/parent-child-layout/parent-child-layout.stories.css [hunk 8]
```diff
@@ -39,16 +35,14 @@ h2 {
...
-  padding: var(--salt-spacing-300);
+.preferences-layout-container {
+  width: 100%;
 }
 
-.parent-child-composite-container .child {
-  max-width: 338px;
-  padding: var(--salt-spacing-200);
+.preferences-layout-parent-view {
+  width: 160px;
```
[the remark this answers is not in the document]
Comment (lukeac123) [reply packages/lab/stories/parent-child-layout/parent-child-layout.stories.css:43]: fair point, its what's in the design spec but could add set to min-content and add a max-width if you think that's preferred ?

File: packages/lab/src/parent-child-layout/ParentChildLayout.tsx (hunk 7 above)
Comment (lukeac123) [reply packages/lab/src/parent-child-layout/ParentChildLayout.tsx]: removed - thanks

File: site/src/examples/parent-child-layout/PreferencesDialog.tsx [hunk 9]
```diff
@@ -207,19 +211,43 @@ export const PreferencesLayout = () => {
...
+        open={open}
+        onOpenChange={onOpenChange}
+        className={styles.dialog}
+      >
+        <DialogCloseButton onClick={handleClose} />
+        <DialogContent>
+          <ParentChildLayout
+            gap={1}
+            collapsedView={collapsedView}
+            // collapseAtBreakpoint="sm"
```
Comment (lilyvc) [inline site/src/examples/parent-child-layout/PreferencesDialog.tsx:243]:
```suggestion
            collapseAtBreakpoint="sm"
```

File: packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx [hunk 10]
```diff
@@ -3,88 +3,58 @@ import * as parentChildStories from "@stories/parent-child-layout/parent-child-l
...
-  describe("WHEN in stacked view", () => {
-    it("THEN it should only display the parent by default", () => {
-      cy.mount(<Collapsed />);
+describe("WHEN collapsableAtbreakpoint is provided", () => {
+  it(
+    "THEN it should be uncollapsed when the current breakpoint is greater than the target breakpoint",
+    { viewportHeight: 900, viewportWidth: 1920 },
+    () => {
+      cy.mount(<Default collapseAtBreakpoint="lg" />);
+      cy.get(".saltParentChildLayout-collapsed").should("not.be.visible");
```
Comment (joshwooding) [inline packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx:22]: Is there no better way than to look for classes? Is it possible to look for the content of what's showing

File: packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx [hunk 11]
```diff
@@ -3,88 +3,58 @@ import * as parentChildStories from "@stories/parent-child-layout/parent-child-l
...
+    }
+  );
+
+  it(
+    "THEN it should be collapsed when the current breakpoint less than or equal to the target breakpoint",
+    { viewportHeight: 900, viewportWidth: 1280 },
+    () => {
+      cy.mount(<Default collapseAtBreakpoint="lg" />);
+      cy.get(".saltParentChildLayout-collapsed").should("be.visible");
```
Comment (joshwooding) [inline packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx:31]: Same comment here

File: packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx [hunk 12]
```diff
@@ -3,88 +3,58 @@ import * as parentChildStories from "@stories/parent-child-layout/parent-child-l
...
+    }
+  );
 
-    it("THEN it should change to the child view when the button is clicked", () => {
-      cy.mount(<Collapsed />);
+  it(
+    "THEN it should display the Parent component when passed as the collapsedViewProp",
+    { viewportHeight: 900, viewportWidth: 600 },
+    () => {
+      cy.mount(<Default collapsedView="child" />);
```
Comment (joshwooding) [inline packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx:53]: I'm not sure how this is working? If the collapsed view is child and the layout is collapsed surely child should be shown?

File: packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx [hunk 13]
```diff
@@ -3,88 +3,58 @@ import * as parentChildStories from "@stories/parent-child-layout/parent-child-l
...
+    { viewportHeight: 900, viewportWidth: 1280 },
+    () => {
+      cy.mount(<Default collapseAtBreakpoint="lg" />);
+      cy.get(".saltParentChildLayout-collapsed").should("be.visible");
+    }
+  );
+});
 
-      cy.get(".saltFlexItem").should("have.length", 1);
+describe("WHEN collapsable", () => {
```
Comment (joshwooding) [inline packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx:36]: Is there a way to disable collapsing?

File: packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx
Comment (joshwooding) [inline packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx:1]: We should test if custom breakpoints work

File: packages/lab/stories/parent-child-layout/parent-child-layout.stories.tsx [hunk 14]
```diff
@@ -42,31 +46,32 @@ export const Default: StoryFn<typeof ParentChildLayout> = (args) => (
...
-        <ToggleButton value="child" onClick={handleChild}>
-          Child
-        </ToggleButton>
-      </ToggleButtonGroup>
+      <StackLayout align="center" gap={1}>
+        <Label>Collapsed View: </Label>
+        <ToggleButtonGroup defaultValue="parent">
+          <ToggleButton
+            value="parent"
+            onClick={() => setCollapsedView("parent")}
```
Comment (joshwooding) [inline packages/lab/stories/parent-child-layout/parent-child-layout.stories.tsx:66]: This should be using an onChange on ToggleButtonGroup

File: packages/lab/stories/parent-child-layout/parent-child-layout.stories.tsx [hunk 15]
```diff
@@ -42,31 +46,32 @@ export const Default: StoryFn<typeof ParentChildLayout> = (args) => (
...
+          >
+            Parent
+          </ToggleButton>
+          <ToggleButton value="child" onClick={() => setCollapsedView("child")}>
```
Comment (joshwooding) [inline packages/lab/stories/parent-child-layout/parent-child-layout.stories.tsx:70]: Same here

File: packages/lab/stories/parent-child-layout/parent-child-layout.stories.tsx [hunk 16]
```diff
@@ -119,7 +125,7 @@ ReducedMotion.args = {
   child,
 };
 
-// Preferences layout
+// Preferences Dialog
```
Comment (joshwooding) [inline packages/lab/stories/parent-child-layout/parent-child-layout.stories.tsx:128]: This should be moved under patterns later

File: site/src/examples/parent-child-layout/Collapsed.tsx [hunk 17]
```diff
@@ -9,32 +14,34 @@ const parent = <div className={styles.parentContent}>Parent</div>;
...
-          Child
-        </ToggleButton>
-      </ToggleButtonGroup>
+      <StackLayout align="center" gap={1}>
+        <Label>Collapsed View: </Label>
+
+        <ToggleButtonGroup defaultValue="parent">
+          <ToggleButton
+            value="parent"
+            onClick={() => setCollapsedView("parent")}
```
Comment (joshwooding) [inline site/src/examples/parent-child-layout/Collapsed.tsx:36]: This needs to be updated to use ToggleButtonGroup's onChange

File: site/src/examples/parent-child-layout/Collapsed.tsx [hunk 18]
```diff
@@ -9,32 +14,34 @@ const parent = <div className={styles.parentContent}>Parent</div>;
...
+          >
+            Parent
+          </ToggleButton>
+          <ToggleButton value="child" onClick={() => setCollapsedView("child")}>
```
Comment (joshwooding) [inline site/src/examples/parent-child-layout/Collapsed.tsx:40]: Same here

File: site/src/examples/parent-child-layout/PreferencesDialog.tsx [hunk 19]
```diff
@@ -154,22 +157,23 @@ export const PreferencesLayout = () => {
     },
   ];
 
-  const [currentView, setCurrentView] = useState<
-    StackedViewElement | undefined
-  >();
+  const [collapsedView, setcollapsedView] = useState<"child" | "parent">(
```
Comment (joshwooding) [inline site/src/examples/parent-child-layout/PreferencesDialog.tsx:160]:
```suggestion
  const [collapsedView, setCollapsedView] = useState<"child" | "parent">(
```
This should be camel case

File: packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx (hunk 13 above)
Comment (lukeac123) [reply packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx:36]: no there isn't, will rename - unless you think there should be the functionality to disable collapsing ?

File: packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx (hunk 12 above)
Comment (lukeac123) [reply packages/lab/src/__tests__/__e2e__/parent-child-layout/ParentChildLayout.cy.tsx:53]: collapsedView - meaning that is the view that is collapsed, so when child is provided, the child is collapsed therefore showing the parent. Going to rename this to visibleView to remove ambiguity

File: packages/lab/src/parent-child-layout/ParentChildLayout.tsx [hunk 20]
```diff
@@ -1,39 +1,30 @@
...
    * Breakpoint at which the parent and child will stack.
    */
   collapseAtBreakpoint?: keyof Breakpoints;
   /**
-   * Change element that is displayed when in staked view.
-   */
-  collapsedViewElement?: StackedViewElement;
-  /**
-   * Disable all animations.
+   * Change element that is displayed when in stacked view.
```
Comment (lukeac123) [inline packages/lab/src/parent-child-layout/ParentChildLayout.tsx:21]:
```suggestion
   * View that is displayed when in a collapsed state.
```

**Review by lilyvc — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Pull request merged**